Core/Meta/BebopSchema.cs

```diff
@@ -186,18 +190,26 @@ public List<SpanException> Validate()
                 }
                 if (definition is EnumDefinition ed)
                 {
-                    var values = new HashSet<uint>();
+                    var values = new HashSet<BigInteger>();
                     var names = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
                     foreach (var field in ed.Members)
                     {
                         if (!ed.IsBitFlags && values.Contains(field.ConstantValue))
                         {
-                            errors.Add(new InvalidFieldException(field, "Enum value must be unique"));
+                            errors.Add(new InvalidFieldException(field, "Enum value must be unique if the enum is not a [flags] enum"));
                         }
                         if (names.Contains(field.Name))
                         {
                             errors.Add(new DuplicateFieldException(field, ed));
                         }
+                        if (!BaseTypeHelpers.InRange(ed.BaseType, field.ConstantValue))
+                        {
+                            var min = BaseTypeHelpers.MinimumInteger(ed.BaseType);
+                            var max = BaseTypeHelpers.MaximumInteger(ed.BaseType);
+                            errors.Add(new InvalidFieldException(field,
```

Comment:
I guess you don't use `InRange`? (Hard to tell from the git view)

Reply:
It's one line up from here :)